Add docs about the App preview feature (#2756)
codemagic-ci-cd/codemagic-docs: content/partials/app-preview-builds.md

---
---

> Codemagic allows you to preview your `.app` artifact on an iOS simulator and interact with the simulator via a web browser. You can easily test the functionalities of your app as well as system notifications and flows that require location data on different simulator devices without requiring a physical machine.

Reviewer: ```suggestion
Codemagic allows you to preview your unsigned application (`.app` artifact) on an iOS simulator and interact with the simulator via a web browser. You can easily test the functionalities of your app as well as system notifications and flows that require location data on different simulator devices without requiring a physical phone.
```

> Codemagic allows you to preview your unsigned `.app` artifact on an iOS simulator and interact with the simulator via a web browser. You can easily test the functionalities of your app as well as system notifications and flows that require location data on different simulator devices without requiring a physical device.

Reviewer 2: Apps can be signed too, it is just unlikely that anyone would go through the trouble.

```suggestion
Codemagic allows you to preview your `.app` artifact on an iOS simulator and interact with the simulator via a web browser. You can easily test the functionalities of your app as well as system notifications and flows that require location data on different simulator devices without requiring a physical device.
```

{{<notebox>}}
Note that this feature is available for **teams** on the Pay as you go billing plan by default and can be enabled for teams on the annual plan on request.
{{</notebox>}}

## Running preview builds

You can preview any `.app` artifact built in Codemagic that targets the iOS simulator. For such artifacts, there is a **Quick Launch** button available next to the artifact name on the build overview page.

Clicking **Quick Launch** displays a configuration popup for selecting the simulator device and runtime. Once you click **Start**, Codemagic starts a preview build during which the simulator is booted and the app is installed on the simulator. 

> When the simulator is ready for use, you'll see the URL to access the simulator printed in the logs under the "Running iOS simulator" step as well as a link to open the simulator on the left panel of the build overview page. The simulator session remains active for a maximum of **20 minutes** or until you cancel the build. 

Reviewer 2: As the link on sidebar is far more user friendly I'd try to highlight it over the URL in logs, which isn't even formatted as a proper href, rendering it difficult to use. For starters I presume that even documenting the sidebar link/button first would be enough.

Author: As the sidebar link may also be hard to spot as it appears there only when the simulator becomes available, I think it's still a good idea to point out that the URL is also available in the logs.

> Once the session ends, the virtual machine running the build is wiped clean and destroyed.

Reviewer: ```suggestion
Once the session ends you can re-run it if needed.
```

## Build minutes usage

> Preview builds run on a macOS VM and consume build minutes at the same per-minute rate as regular builds. If your team is on a Pay as you go plan, the minutes used for app previews will be added to your macOS minutes count.

Reviewer: ```suggestion
Preview builds run on a macOS and consume build minutes at the same per-minute rate as regular builds. If your team is on a Pay as you go plan, the minutes used for app previews will be added to your macOS minutes count.
```
